tools/lxdclient/client_image.go

```diff
@@ -92,56 +97,107 @@ func (p *progressContext) copyProgress(progress string) {
 // @param copyProgressHandler: a callback function. If we have to download an
 // image, we will call this with messages indicating how much of the download
 // we have completed (and where we are downloading it from).
-func (i *imageClient) EnsureImageExists(series string, sources []Remote, copyProgressHandler func(string)) error {
+func (i *imageClient) EnsureImageExists(series string, sources []Remote, copyProgressHandler func(string)) (string, error) {
 	// TODO(jam) Find a way to test this, even though lxd.Client can't
 	// really be stubbed out because CopyImage takes one directly and pokes
 	// at private methods so we can't easily tweak it.
-	name := i.ImageNameForSeries(series)
 
-	lastErr := errors.New("image not imported!")
+	// NOTE(axw) architecture should be passed into this function, as
+	// indicated by the TODO in the function's doc comment. When it is,
+	// use get rid of this.
+	const arch = "amd64"
```

Comment:
OK, seems that there's some magic somewhere that filters on the client's arch automatically.